lnd_methods/offchain/estimate_route_fee.js

```diff
@@ -0,0 +1,91 @@
+const asyncAuto = require('async/auto');
+const {returnResult} = require('asyncjs-util');
+const {isStringLiteral, isNumericLiteral} = require('typescript');
+
+/** Estimate routing fees based on an invoice.
+
+  Requires `offchain:read` permission
+
+  {
+    lnd: <Authenticated LND API Object>
+    request: <BOLT 11 Payment Request String>
+    timeout: <Route Fee Estimate Timeout in Milliseconds Number> 
+  }
+
+  @returns via cbk or Promise
+  {
+    routing_fee_msat: <Route Fee Millitokens String>,
+    time_lock_delay: <Time Lock Block Height Delay String>
+    failure_reason: <Failure Reason String>
+  }
+*/
+module.exports = (args, cbk) => {
+  return new Promise((resolve, reject) => {
+    return asyncAuto({
+      // Check arguments
+      validate: cbk => {
+        if (!args.lnd || !args.lnd.router) {
+          return cbk([400, 'ExpectedAuthenticatedLndForRouteFeeEstimate']);
+        }
+        
+        if (!args.request || !isStringLiteral(args.request)) {
+          return cbk([400, 'ExpectedPaymentRequestStringForRouteFeeEstimate']);
+        }
+
+        if (!args.timeout || !isNumericLiteral(args.timeout)) {
+          return cbk([400, 'ExpectedTimeoutNumberForRouteFeeEstimate'])
+        }
+
+        if (args.timeout > 86400000) {
+          return cbk([400, 'ExpectedTimeoutLessThanOneDayForRouteFeeEstimate']);
+        }
+
+        if(args.timeout < 1) {
+          return cbk([400, 'ExpectedTimeoutGreaterThanZeroForRouteFeeEstimate']);
+        }
+
+        return cbk();
+      },
+
+      // Estimate route fee and return a successful routing fee and time lock delay
+      estimateRouteFee: ['validate', ({}, cbk) => {
+        const request = args.request;
+        const timeout = args.timeout;
+        return args.lnd.router.estimateRouteFee({request, timeout}, (err, res) => {  
+            if (!!err) {
+              return cbk([503, 'UnexpectedEstimateRouteFeeError', {err}]);
+            }
+  
+            if (!res) {
+              return cbk([503, 'ExpectedRouteFeeResponse']);
+            }
+  
+            if (!res.routing_fee_msat) {
+              return cbk([503, 'ExpectedRoutingFeeMsatInRouteFeeResponse']);
+            }
+
+            if (!res.time_lock_delay) {
+                return cbk([503, 'ExpectedTimeLockDelayInRouteFeeResponse']);
+            }
+
+            if (!res.failure_reason) {
+                return cbk([503, 'ExpectedFailureReasonInRouteFeeResponse']);
+            }
+
+            if (res.failure_reason !== 'FAILURE_REASON_NONE') {
```

Comment:
this could be more like return cbk([404, 'EstimateRouteFindingFailed', {failure: res.failure:reason}]) and not mess with the reason to allow for the caller to just say oh there isn't a path that could be found